src/lib/tvgBezier.cpp

```diff
@@ -74,8 +90,10 @@ float bezLength(const Bezier& cur)
     auto len = _lineLength(cur.start, cur.ctrl1) + _lineLength(cur.ctrl1, cur.ctrl2) + _lineLength(cur.ctrl2, cur.end);
     auto chord = _lineLength(cur.start, cur.end);
 
-    if (fabs(len - chord) > FLT_EPSILON) {
+    if (fabs(len - chord) > BEZIER_EPSILON) {
         bezSplit(cur, left, right);
+        if (bezEqual(left, right) || bezEqual(cur, right) || bezEqual(cur, left))
```

Comment:
we can adjust BEZIER_EPSION value, then do we really need bezEqual() comparison?
also, it looks  (left - cur),  (cur - right) 2 time compare is necessary, not 3 times.

Reply:
For the epsilon value of 1e-5 I can easily find an example that will crash without bezEqual() - I modified BezierLength.cpp. In this example the difference between 'len' and 'chord' variables (the current termination condition) is still > BEZ_EPS - the value is ~3e-5. I can change the epsilon value to be 1e-4 and that will solve the problem in this case.
 
But I can keep changing the Bezier curve from the example in such a way, that it will keep crashing even for the epsilon value > 1e-4 - e.g. a curve that is defined by the coordinates x,y > 3000.

While I don't think it's a common case, I'm not sure that it's impossible to find a more realistic example, that will crash as well for eps = 1e-4. It's the case where we need to balance between performance and the accuracy. If you are ok with epsilon value = 1e-4, I would 'take a risk' and omit double calling of bezEqual() fun (you are right, calling it twice is enough).